### ISSUE TITLE
Retrieve result using just `taskUUID` instead of holding onto the `AsyncResult` object?

### ISSUE THREAD

**Issue by rakeshbala** [RichardKnop/machinery#186]
I am planning to use `machinery` to convert a few REST endpoints that take a long time to respond to return immediately by offloading the work to task queues.
I was thinking of returning a result polling URL from each of this endpoints to query the result.

From my understanding of how machinery works this would involve me holding onto the AsyncResult object returned from each task dispatch call and managing them in a global data structure somewhere. And when the user queries the result URL (which will have the `taskUUID` as the parameter) I will resolve the result and send it back.

However, it would be much easier (for me) if there was a way to retrieve the result using just the `taskUUID` (if the task hasn't expired) since I don't have to maintain all that state (and since that state is already being maintained by machinery).

In other words is there a way to construct an `AsyncResult` object from a valid `taskUUID` so that I don't have to hold onto the AsyncResult from the task dispatch call?

 #TIA

**Comment by rakeshbala** [comment]
Alright. I realized that I can re-create the `AsyncResult` if I specify a UUID for the signature and then use a signature with same UUID while creating the new AsyncResult.

```
    uuid := uuid.NewV4()
    signature := &tasks.Signature{
        Name: "add",
        Args: []tasks.Arg{
            {
                Type:  "int64",
                Value: 1,
            },
            {
                Type:  "int64",
                Value: 2,
            },
        },
        UUID: uuid.String(),
    }

    _, err = server.SendTask(signature)
```

and then later 

```
    sig := &tasks.Signature{
        Name: "add",
        UUID: uuid.String(),
    }
   asyncResult := backends.NewAsyncResult(sig, server.GetBackend())
```

gives me the right result. 

I'll leave the issue open for a couple of days to see if you have any comments.

**Comment by RichardKnop** [comment]
Yes, this is correct approach. The only attribute from `Signature` object inside `AsyncResult` needed to retrieve the task state from the backend is UUID. So you can pass in otherwise empty signature - just make sure to set UUID and you should be able to poll results.
